app/models/encrypted_data.rb

```diff
@@ -13,11 +13,19 @@ def [](key)
     value = @data[key]
     iv = decode(value[:iv]) unless value[:iv].nil?
 
-    data_hash(iv, decode(value[:data]))
+    if key == :custom_attr
+      data_hash_label(iv, value[:label], decode(value[:data]))
+    else
+      data_hash(iv, decode(value[:data]))
+    end
   end
 
-  def []=(key, data:, iv:)
-    @data[key] = data_hash(encode(iv), encode(data))
+  def []=(key, data:, iv:, label: nil)
+    @data[key] = if label&.present?
```

Comment:
`if label` would probably work just as well, however this won't be needed anymore anyway when you look at the next comment :)